source/common/http/conn_pool_grid.h

```diff
@@ -206,9 +212,9 @@ class ConnectivityGrid : public ConnectionPool::Instance,
   // that specifies HTTP/3 and HTTP/3 is not broken.
   bool shouldAttemptHttp3();
 
-  // Creates the next pool in the priority list, or nullptr if all pools have been created.
-  // TODO(alyssawilk) replace this now we have explicit pools.
-  virtual ConnectionPool::Instance* createNextPool();
+  // Creates the requested pool
```

Comment:
nit: How about "Returns the specified pool, which will be created if necessary"

Reply:
done